ext/xml/xml.c

```diff
@@ -1507,18 +1507,18 @@ PHP_FUNCTION(xml_parser_set_option)
 			parser->parsehuge = zend_is_true(value);
 			break;
 		/* Integer option */
-		case PHP_XML_OPTION_SKIP_TAGSTART:
+		case PHP_XML_OPTION_SKIP_TAGSTART: {
 			/* The tag start offset is stored in an int */
-			/* TODO Improve handling of values? */
-			parser->toffset = zval_get_long(value);
-			if (parser->toffset < 0) {
+			zend_long value_long = zval_get_long(value);
```

Comment:
The TODO still applies, as when I added it I wanted to go back and make this use `zend_try_get_long()` (or however this function is called) to deal with objects, arrays, resources, and strings better.

Reply:
Ah okay, I thought this was about the incomplete check, but what you say makes sense indeed!